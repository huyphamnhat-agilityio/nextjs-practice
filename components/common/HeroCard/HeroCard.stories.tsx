import type { Meta, StoryObj } from "@storybook/react";

// Components
import HeroCard from ".";
import { BlackBoardIcon, MortarboardIcon, TelescopeIcon } from "../Icons";

const meta: Meta<typeof HeroCard> = {
  title: "Components/common/HeroCard",
  component: HeroCard,
  argTypes: {
    icon: {
      description: "Icon of the hero card",
      control: {
        disable: true,
      },
    },
    iconBackgroundColor: {
      description: "Background color of the icon button",
      control: {
        type: "select",
        options: ["primary", "secondary", "default"],
      },
    },
    title: {
      description: "Title of the hero card",
    },
    description: {
      description: "Description of the hero card",
    },
  },
  args: {
    icon: <BlackBoardIcon />,
    title: "Expert instruction",
    description:
      "The gradual accumulation of information about atomic and small-scale behaviour...",
    iconBackgroundColor: "primary",
  },
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof HeroCard>;

export const Primary: Story = {};
