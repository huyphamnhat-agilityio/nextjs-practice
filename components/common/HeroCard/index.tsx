import { Card, CardBody, CardHeader, Divider } from "@nextui-org/react";
import React from "react";

// Components
import { Button } from "../Button";

export type HeroCardProps = {
  icon: JSX.Element;
  iconBackgroundColor: "primary" | "secondary" | "secondary-100";
  title: string;
  description: string;
};
const HeroCard = ({
  icon,
  iconBackgroundColor,
  title,
  description,
}: HeroCardProps) => {
  return (
    <Card className="max-w-xs sm:max-w-82 px-10 py-8.75  flex flex-col gap-5 rounded-none">
      <CardHeader className="flex flex-col gap-5 items-start p-0">
        <div
          className={`text-white flex justify-center items-center px-4.75 py-5.5 w-auto h-auto rounded-2.5 bg-${iconBackgroundColor}`}
        >
          {icon}
        </div>
        <h3 className="text-2xl font-bold text-foreground">{title}</h3>
      </CardHeader>

      <Divider className="w-12.5 h-0.5 bg-primary" />

      <CardBody className="p-0">
        <p className="text-sm text-foreground-100">{description}</p>
      </CardBody>
    </Card>
  );
};

export default HeroCard;
