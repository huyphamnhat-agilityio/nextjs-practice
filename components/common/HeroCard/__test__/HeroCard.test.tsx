import { render } from "@testing-library/react";

// Components
import HeroCard, { HeroCardProps } from "..";

describe("HeroCard test cases", () => {
  const mockProps: HeroCardProps = {
    icon: <></>,
    iconBackgroundColor: "primary",
    title: "Test Title",
    description: "Test Description",
  };
  const setup = (props: HeroCardProps) => render(<HeroCard {...props} />);

  it("should render correctly", () => {
    const { container } = setup(mockProps);

    expect(container).toMatchSnapshot();
  });
});
