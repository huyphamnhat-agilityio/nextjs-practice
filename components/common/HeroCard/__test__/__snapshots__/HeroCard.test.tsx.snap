// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard test cases should render correctly 1`] = `
<div>
  <div
    class="relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none px-10 py-8.75 w-82 flex flex-col gap-5 rounded-none"
    tabindex="-1"
  >
    <div
      class="z-10 w-full justify-start shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large flex flex-col gap-5 items-start p-0"
    >
      <div
        class="text-white flex justify-center items-center px-4.75 py-5.5 w-auto h-auto rounded-2.5 bg-primary"
      />
      <h3
        class="text-2xl font-bold text-foreground"
      >
        Test Title
      </h3>
    </div>
    <hr
      class="shrink-0 border-none w-12.5 h-0.5 bg-primary"
      role="separator"
    />
    <div
      class="relative flex w-full flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased p-0"
    >
      <p
        class="text-sm text-foreground-100"
      >
        Test Description
      </p>
    </div>
  </div>
</div>
`;
